components/pageLoader.tsx

```diff
@@ -0,0 +1,114 @@
+import cachedData from "../cachedData.json";
+import Page from "./Page";
+import { JSObjectFromJSON } from "../src/dataProcessing";
+import { getJsonStructure } from "evergreen-org-crawler/src/index"
+import config from "evergreen-org-crawler/config.json"
+import { useEffect, useState } from "react";
+import CircularProgress from '@mui/material/CircularProgress';
+import Box from '@mui/material/Box';
+
+enum Mode{
+	Frontend,
+	StandaloneBackend,
+	IntegratedBackend
+}
+
+//TODO: Move to config file
+const mode: Mode = Mode.IntegratedBackend
+
+function getProperty<T, K extends keyof T>(o: T, propertyName: K): T[K] {
+	return o[propertyName];
+}
```

Comment:
This function seems unnecessary. 
I guess it was probably written to fight typescript errors, but if you have input arguments `o:T` and `propertyName:K` you should always be able to do `o[propertyName]` outside the function